frontend/src/features/documents/documentUpload/DocumentUploadContainer.tsx

```diff
@@ -35,8 +40,16 @@ export const DocumentUploadContainer: React.FunctionComponent<IDocumentUploadCon
     fetch();
   }, [getDocumentTypes, isMounted]);
 
-  const onDocumentTypeChange = (param: any) => {
-    // Todo: This will guide the retrieval of the metadata types
+  const onDocumentTypeChange = async (changeEvent: ChangeEvent<HTMLInputElement>) => {
+    const documentTypeId = Number(changeEvent.target.value);
+    const mayanDocumentTypeIdId = documentTypes.find(x => x.id === documentTypeId)?.mayanId;
+    if (mayanDocumentTypeIdId) {
+      const axiosResponse = await getDocumentTypeMetadata(mayanDocumentTypeIdId);
```

Comment:
does this work gracefully if mayan is down?

Reply:
yes the dropdown remains empty if mayan is down, checked it locally.